schema_based_on_diagram.sql

```diff
@@ -0,0 +1,40 @@
+CREATE TABLE patients (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    name VARCHAR(100) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE medical_histories (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    admitted_at TIMESTAMP,
+    patient_id INT NULL REFERENCES patients(id) INT,
+    status VARCHAR(50) NOT NULL
+);
+
+CREATE TABLE invoices (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    total DECIMAL NOT NULL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT NULL REFERENCES medical_histories(id)
+);
+
+CREATE TABLE treatments (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    type VARCHAR(100) NOT NULL,
+    name VARCHAR(50) NOT NULL
+);
+
+CREATE TABLE invoice_items (
+    id BIGSERIAL NOT NULL PRIMARY KEY,
+    unit_price DECIMAL NOT NULL,
+    quantity INT NOT NULL,
+    total_price DECIMAL NOT NULL,
+    invoice_id INT NULL REFERENCES invoices(id),
+    treatment_id INT NULL REFERENCES treatments(id)
+);
+
+CREATE TABLE medical_histories_treatments (
+    medical_history_id INT REFERENCES medical_histories(id),
+    treatment_id INT REFERENCES treatments(id)
```

Comment:
- [ ] Please create indexes on all `Foreign keys`

Reply:
Thanks for the review

Indexes has been created for the foreign keys.